src/codegen.cpp

```diff
@@ -9676,17 +9677,9 @@ void jl_compile_workqueue(
                 // method body. See #34993
                 if (policy != CompilationPolicy::Default &&
                     jl_atomic_load_relaxed(&codeinst->inferred) == jl_nothing) {
-                    src = jl_type_infer(codeinst->def, jl_atomic_load_acquire(&jl_world_counter), 0);
-                    if (src) {
-                        orc::ThreadSafeModule result_m =
-                        jl_create_ts_module(name_from_method_instance(codeinst->def),
-                            params.tsctx, params.DL, params.TargetTriple);
-                        auto decls = jl_emit_code(result_m, codeinst->def, src, src->rettype, params);
-                        if (result_m)
-                            it = params.compiled_functions.insert(std::make_pair(codeinst, std::make_pair(std::move(result_m), std::move(decls)))).first;
-                    }
+                    codeinst = jl_type_infer(codeinst->def, jl_atomic_load_acquire(&jl_world_counter), 0, SOURCE_MODE_FORCE_SOURCE);
```

Comment:
What have we been doing with a randomly invented world value here.
```suggestion
                    codeinst = jl_type_infer(codeinst->def, params.max_world, 0, SOURCE_MODE_FORCE_SOURCE);
```

Reply:
Also needs to check that the assertion above holds, and bail out of this if it fails, before we trigger that assertion:
```
        assert(jl_atomic_load_relaxed(&codeinst->min_world) <= params.min_world &&
               jl_atomic_load_relaxed(&codeinst->max_world) >= params.max_world &&
```

Reply:
I don't really know if this code path makes any sense anymore with the new cache split stuff, but for now, I'll change it to codeinst->max_world.